api/serialization/parquet.go

```diff
@@ -105,10 +104,27 @@ func (d *Parquet) ReadData(uri string) ([][]string, error) {
 	for i := int64(0); i < colCount; i++ {
 		colRaw, err := d.readColumn(pr, i, rowCount)
 		if err != nil {
-			return nil, err
+			fr.Close()
```

Comment:
I think this solution would be more elegant than getting the `ulimit -n` and checking if we reached it or not.